app/grandchallenge/evaluation/tasks.py

```diff
@@ -313,10 +323,18 @@ def set_evaluation_inputs(*, evaluation_pk):
         .prefetch_related("values")
     }
 
+    if evaluation.submission.algorithm_model:
+        extra_filter = {
+            "algorithm_model": evaluation.submission.algorithm_model
+        }
+    else:
+        extra_filter = {}
```

Comment:
Don't you want to set `"algorithm_model__isnull": True` here? Same above for `pending_jobs_extra_filter`

Reply:
Yes, clearly my tests need to be expanded! Working on that now.

Reply:
Suffering from brain fog this morning, but I think the tests now cover all situations. I think the tests could be improved, they are hard to read at the moment, but I couldn't find a nice way to parametrize the different situations...

Reply:
I spent ages yesterday fighting a test that was far too long. The ones in test_tasks are short and clear, and no logic. Perfect. Easy to understand, easy to debug later. If anything `test_eval_exists_for_image_and_model` is too smart - two if statements which really shouldn't be in tests. A better approach may be to use two tests with module scoped fixtures. But, whatever.